lib/chef/provider/windows_task.rb

```diff
@@ -194,26 +196,48 @@ def run_schtasks(task_action, options = {})
 
       def task_need_update?
         return true if (new_resource.command &&
-            @current_resource.command != new_resource.command.tr("'", '"')) ||
-            @current_resource.user != new_resource.user ||
-            @current_resource.run_level != new_resource.run_level ||
-            @current_resource.cwd != new_resource.cwd ||
-            @current_resource.frequency_modifier != new_resource.frequency_modifier ||
-            @current_resource.frequency != new_resource.frequency ||
-            @current_resource.idle_time != new_resource.idle_time ||
-            @current_resource.random_delay != new_resource.random_delay ||
-            @current_resource.execution_time_limit != new_resource.execution_time_limit ||
-            !new_resource.start_day.nil? || !new_resource.start_time.nil?
+            current_resource.command != new_resource.command.tr("'", '"')) ||
+            current_resource.user != new_resource.user ||
+            current_resource.run_level != new_resource.run_level ||
+            current_resource.cwd != new_resource.cwd ||
+            current_resource.frequency_modifier != new_resource.frequency_modifier ||
+            current_resource.frequency != new_resource.frequency ||
+            current_resource.idle_time != new_resource.idle_time ||
+            current_resource.random_delay != new_resource.random_delay ||
+            !new_resource.execution_time_limit.include?(current_resource.execution_time_limit) ||
+            (new_resource.start_day && start_day_updated?) ||
+            (new_resource.start_time && start_time_updated?)
         begin
-          return true if new_resource.day.to_s.casecmp(@current_resource.day.to_s) != 0 ||
-              new_resource.months.to_s.casecmp(@current_resource.months.to_s) != 0
+          return true if new_resource.day.to_s.casecmp(current_resource.day.to_s) != 0 ||
+              new_resource.months.to_s.casecmp(current_resource.months.to_s) != 0
         rescue
           Chef::Log.debug "caught a raise in task_needs_update?"
         end
 
         false
       end
 
+      def start_day_updated?
+        current_day = DateTime.parse(current_resource.start_day)
+        new_day = DateTime.parse(new_resource.start_day)
+        current_day != new_day
+      end
+
+      def start_time_updated?
+        time = DateTime.parse(current_resource.start_time).strftime("%H:%M")
+        time != new_resource.start_time
+      end
+
+      def convert_user_date_to_system_date(date_in_string)
+        Chef::Log.debug "Converting '#{date_in_string}' to system date format"
+        task_script = <<-EOH
+          [Console]::OutputEncoding = [Text.UTF8Encoding]::UTF8
+          ([datetime]"#{date_in_string}").ToString((((((([Globalization.Cultureinfo]::CurrentCulture.DateTimeFormat.ShortDatePattern -replace "dd", "d") -replace "d", "dd") -replace "MM", "M")-replace "M","MM") -replace "yyyy","yy") -replace "y", "yy"))
```

Comment:
Has this been tested against multiple locales? On my system which is in English (United Kingdom) this line would not accept a date formatted in the ShortDatePattern for my locale:

```
PS C:\Users\spreston> get-culture

LCID             Name             DisplayName                                                                                                      
----             ----             -----------                                                                                                      
2057             en-GB            English (United Kingdom)  
```

```
([datetime]"30/08/2017").ToString((((((([Globalization.Cultureinfo]::CurrentCulture.DateTimeFormat.ShortDatePattern -replace "dd", "d") -replace "d", "dd") -replace "MM", "M")-replace "M","MM") -replace "yyyy","yy") -replace "y", "yy"))
Cannot convert value "30/08/2017" to type "System.DateTime". Error: "String was not recognized as a valid DateTime."
At line:1 char:1
+ ([datetime]"30/08/2017").ToString((((((([Globalization.Cultureinfo]:: ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [], RuntimeException
```

But it does take an English (US) format date:
```
([datetime]"08/30/2017").ToString((((((([Globalization.Cultureinfo]::CurrentCulture.DateTimeFormat.ShortDatePattern -replace "dd", "d") -replace "d", "dd") -replace "MM", "M")-replace "M","MM") -replace "yyyy","yy") -replace "y", "yy"))
30/08/2017
```

```
PS C:\Users\spreston> [Globalization.Cultureinfo]::CurrentCulture.DateTimeFormat.ShortDatePattern
dd/MM/yyyy

PS C:\Users\spreston> [Globalization.Cultureinfo]::CurrentCulture.DateTimeFormat


AMDesignator                     : AM
Calendar                         : System.Globalization.GregorianCalendar
DateSeparator                    : /
FirstDayOfWeek                   : Monday
CalendarWeekRule                 : FirstFourDayWeek
FullDateTimePattern              : dd MMMM yyyy HH:mm:ss
LongDatePattern                  : dd MMMM yyyy
LongTimePattern                  : HH:mm:ss
MonthDayPattern                  : d MMMM
PMDesignator                     : PM
RFC1123Pattern                   : ddd, dd MMM yyyy HH':'mm':'ss 'GMT'
ShortDatePattern                 : dd/MM/yyyy
ShortTimePattern                 : HH:mm
SortableDateTimePattern          : yyyy'-'MM'-'dd'T'HH':'mm':'ss
TimeSeparator                    : :
UniversalSortableDateTimePattern : yyyy'-'MM'-'dd HH':'mm':'ss'Z'
YearMonthPattern                 : MMMM yyyy
AbbreviatedDayNames              : {Sun, Mon, Tue, Wed...}
ShortestDayNames                 : {Su, Mo, Tu, We...}
DayNames                         : {Sunday, Monday, Tuesday, Wednesday...}
AbbreviatedMonthNames            : {Jan, Feb, Mar, Apr...}
MonthNames                       : {January, February, March, April...}
IsReadOnly                       : False
NativeCalendarName               : Gregorian Calendar
AbbreviatedMonthGenitiveNames    : {Jan, Feb, Mar, Apr...}
MonthGenitiveNames               : {January, February, March, April...}
```